### PR TITLE
Validate external links

### PR DESCRIPTION
**Pull request by bdemers**
- :fire: Removes all `localhost` links
- ✅ Adds external link validation tests
- ✅ Adds `localhost` link checker
    - Failed [commit](https://travis-ci.org/okta/okta.github.io/builds/299244564#L7911-L7913)

### CODE REVIEW

File: _source/_posts/2017-06-21-what-the-heck-is-oauth.md [hunk 1]
```diff
@@ -256,7 +256,7 @@ When people ask if you support OAuth, you have to clarify what they're asking fo
 
 There's a large surface area with OAuth. With Implicit Flow, there's lots of redirects and lots of room for errors. There's been a lot of people trying to exploit OAuth between applications and it's easy to do if you don't follow recommended Web Security 101 guidelines. For example:
 
-* [Always use CSRF token with the `state` parameter to ensure flow integrity](http://www.twobotechnologies.com/blog/2014/02/importance-of-state-in-oauth2.html)
+* ~Always use CSRF token with the `state` parameter to ensure flow integrity~ (dead link: http://www.twobotechnologies.com/blog/2014/02/importance-of-state-in-oauth2.html)
```
Comment (bdemers) [inline _source/_posts/2017-06-21-what-the-heck-is-oauth.md]: I'm not sure how to handle this one, ideas?

File: _source/_use_cases/mfa/index.md [hunk 2]
```diff
@@ -15,9 +15,7 @@ security to their applications.
 <blockquote class="twitter-tweet" lang="en">
 <p>In all my years working in this industry I&#39;ve seen exactly 2
 effective security technologies stand the test of time: firewalls
-and 2fa</p>&mdash; Chris Rohlf (@chrisrohlf)
-<a href="https://twitter.com/chrisrohlf/status/574639358101352449">
```
Comment (bdemers) [inline _source/_use_cases/mfa/index.md]: A few places online link to this one, but the link is dead

File: _source/_posts/2017-06-21-what-the-heck-is-oauth.md (hunk 1 above)
Comment (jmelberg-okta) [reply _source/_posts/2017-06-21-what-the-heck-is-oauth.md]: @mraible Can we remove this link?

File: _source/_posts/2015-06-11-trustpage.md [hunk 3]
```diff
@@ -6,11 +6,11 @@ tags: [trust, trust pages, trust relationship, product, engineering, customer su
...
+I recently read an excellent article (no longer available: http://www.viabilify.com/blog/trust) about how amazing products shape the **trust relationship** with customers. I think great products are the _first step_ in building a trust relationship. And like other aspects of the product that are derived from the product but are not physically part of it, the trust relationship is now more important than ever before.
 
 * * *
 
->When you use a product, every engagement with that product has a direct correlation with your perception of the value of that product. — From [Product Loyalty Follows Trust Like Form Follows Function](http://www.viabilify.com/blog/trust)
+>When you use a product, every engagement with that product has a direct correlation with your perception of the value of that product. — From 'Product Loyalty Follows Trust Like Form Follows Function' (no longer available: http://www.viabilify.com/blog/trust)
```
Comment (jmelberg-okta) [inline _source/_posts/2015-06-11-trustpage.md]: I wonder if we can completely remove this section + the one above. Or, lets disable the external link checker to look at this page.

File: _source/_data/authors.yml [hunk 4]
```diff
@@ -21,7 +21,6 @@ jon_todd:
   avatar: avatar-jon_todd.jpg
   email: jtodd@okta.com
   web: http://www.jontodd.com
-  twitter: https://twitter.com/jonrtodd
```
Comment (jmelberg-okta) [inline _source/_data/authors.yml]: https://twitter.com/JonToddDotCom

Comment (bdemers) [reply _source/_data/authors.yml]: 👍

File: _source/_posts/2017-06-21-what-the-heck-is-oauth.md (hunk 1 above)
Comment (mraible) [reply _source/_posts/2017-06-21-what-the-heck-is-oauth.md]: @jmelberg-okta Yes, you can remove.

File: _source/_posts/2017-08-09-jax-rs-vs-spring-rest-endpoints.md [hunk 5]
```diff
@@ -224,7 +224,7 @@ We’ll use the same model and DAO for the JAX-RS example, all we’re going to
...
 
-I picked Jersey for a couple reasons, mostly though it was because I already knew how get simple dependency injection working without jumping through any hoops, we are comparing this to Spring after all. Apache Shiro has an [example](https://github.com/apache/shiro/blob/master/samples/jaxrs/) that runs the same code on [Jersey](https://jersey.java.net/), [RestEasy](http://resteasy.jboss.org/), and [Apache CXF](https://cxf.apache.org/), if you’re interested in seeing a portable example.
+I picked Jersey for a couple reasons, mostly though it was because I already knew how get simple dependency injection working without jumping through any hoops, we are comparing this to Spring after all. Apache Shiro has an [example](https://github.com/apache/shiro/blob/master/samples/jaxrs/) that runs the same code on [Jersey](https://jersey.java.net/), [RestEasy](http://resteasy.jboss.org/), and [Apache CXF](http://cxf.apache.org/), if you’re interested in seeing a portable example.
```
Comment (bdemers) [inline _source/_posts/2017-08-09-jax-rs-vs-spring-rest-endpoints.md]: http://cxf.apache.org/ should still be https://cxf.apache.org/

**Comment by bdemers** [comment]
+1

**Pull request merged**